### PR TITLE
Apa documentation

### PR DESCRIPTION
**Pull request by guptasu**
We missed documenting APA when we did the work.

### CODE REVIEW

File: mixer/doc/adapters.md [hunk 1]
```diff
@@ -84,18 +84,34 @@ Every template also has two additional properties associated with it:
...
-associated instances. The `template_variety` also determines under which of the core Mixer behaviors, check report or
-quota, the `instances` for the templates should be created and dispatched to adapters.
+* **Template Variety**: Every template has a specific `template_variety` which can be either Check, Report, Quota or
+AttributeGenerator. The template and its variety determine the signatures of the methods that adapters must implement
+for consuming the associated instances. The `template_variety` also determines under which of the core Mixer behaviors,
+the `instances` for the templates should be created and dispatched to adapters.
+For example:
+    * *Check* template variety instances are created and dispatched only during Mixer's Check API call.
+    * *Report* template variety instances are created and dispatched only during Mixer's Report API call.
+    * *Quota* template variety instances are created and dispatched only during Mixer's Quota API call.
```
Comment (geeknoid) [inline mixer/doc/adapters.md]: There is no QUota API. Quota is part of the CHeck API

File: mixer/doc/adapters.md [hunk 2]
```diff
@@ -84,18 +84,34 @@ Every template also has two additional properties associated with it:
...
+    * *AttributeGenerator* template variety instances are created and dispatched to adapters for all of Mixer's
+Check, Report and QUOTA API calls. The processing of these templates happen during the supplementary attribute generation
```
Comment (geeknoid) [inline mixer/doc/adapters.md]: No QUOTA

File: mixer/doc/adapters.md [hunk 3]
```diff
@@ -84,18 +84,34 @@ Every template also has two additional properties associated with it:
...
+phase which happens before the processing any other variety of templates. Adapters that handle `AttributeGenerator`
```
Comment (geeknoid) [inline mixer/doc/adapters.md]: the processing -> processing

File: mixer/doc/adapters.md (hunk 1 above)
Comment (guptasu) [reply mixer/doc/adapters.md]: D

File: mixer/doc/adapters.md (hunk 2 above)
Comment (guptasu) [reply mixer/doc/adapters.md]: D

File: mixer/doc/adapters.md (hunk 3 above)
Comment (guptasu) [reply mixer/doc/adapters.md]: D

**Comment by guptasu** [comment]
@douglas-reid @geeknoid re-lgtm please

**Pull request merged**